### ISSUE TITLE
Support numeric fields to search in

### ISSUE THREAD

**Issue by AlexanderHofmeister** [typesense/typesense#72]
## Description

According to the API there is no way to sort by fields with type string, e.g the name of a person
Imagine I want to sort all employees by last name. 

## Steps to reproduce
Create collection: 
`{
         "name": "employees",
         "fields": [
           {"name": "name", "type": "string"},
          {"name": "age", "type": "float"}
         ],
         "default_sorting_field": "age"
       }`

Import data:
`{"name":"Stark", "age": 45}
{"name":"Captain Marvel", "age":22}`

Query: `http://localhost:8108/collections/employees/documents/search?q=22&query_by=age`

## Expected Behavior

no error message

## Actual Behavior

Getting an error message: 

{
    "message": "Field `age` should be a string or a string array."
}
## Metadata

**Typsense Version**: 0.10.1

**OS**: Windows 10

**Comment by kishorenc** [comment]
Duplicate of https://github.com/typesense/typesense/issues/73

**Comment by stephanoparaskeva** [comment]
This is crazy. I have data that has no numeric keys:
```JavaScript
const schema = {
  name: 'roles',
  num_documents: 0,
  fields: [
    { name: 'id', type: 'string', facet: false },
    { name: 'title', type: 'string', facet: false },
  ],
  default_sorting_field: 'id',
};
```
> Error must be float or int32

But after changing it to

```JavaScript
const schema = {
  name: 'roles',
  num_documents: 0,
  fields: [
    { name: 'id', type: 'int32', facet: false },
    { name: 'title', type: 'string', facet: false },
  ],
  default_sorting_field: 'id',
};
```
>Error id field should be a string

**Comment by kishorenc** [comment]
Yes there are two different issues here. 

A) the id field must indeed be a string
B) default sorting field is meant to provide a logical sorting order for data in terms of popularity so it cannot be a string. In fact, Typesense does not support sorting by string at all so it's consistent with that.

Happy to clarify further if I've misunderstood your requirement.

**Comment by stephanoparaskeva** [comment]
> Yes there are two different issues here.
> 
> A) the id field must indeed be a string
> B) default sorting field is meant to provide a logical sorting order for data in terms of popularity so it cannot be a string. In fact, Typesense does not support sorting by string at all so it's consistent with that.
> 
> Happy to clarify further if I've misunderstood your requirement.

I can see the issues, but they are circular. If one has data that has no 'sortable' data, then they cannot create a collection for the data. My data should be searchable even if it's not sortable. It's impossible for me to create a TypeSense collection/index unless I change my database structure?

**Comment by kishorenc** [comment]
Typesense needs some field to sort results by. This is to ensure that when you make the same search query twice, the results are ordered in a consistent way. Instead of requiring a `default_sorting_field`, we could order the results in the same order in which they were inserted. If you don't really have a field to nominate as the default sorting field, you can just use a constant value for this field and we will use the inserted order of documents to determine sort order. 

However, the `default_sorting_field` sorting field also serves another useful purpose.

When a word in a search query matches multiple possible words (either because of a typo or during a prefix search), this parameter is used to rank such equally matching tokens. For e.g. both "john" and "joan" are 1-typo away from "jofn". Similarly, in a prefix search, both "apple" and "apply" would match the prefix "app". The `default_sorting_field` tell Typesense which of the two equally-matched word that we should prioritize. That's why we insist of `default_sorting_field` -- by being a little opinionated.

**Comment by stephanoparaskeva** [comment]
> Typesense needs some field to sort results by. This is to ensure that when you make the same search query twice, the results are ordered in a consistent way. Instead of requiring a `default_sorting_field`, we could order the results in the same order in which they were inserted. If you don't really have a field to nominate as the default sorting field, you can just use a constant value for this field and we will use the inserted order of documents to determine sort order.
> 
> However, the `default_sorting_field` sorting field also serves another useful purpose.
> 
> When a word in a search query matches multiple possible words (either because of a typo or during a prefix search), this parameter is used to rank such equally matching tokens. For e.g. both "john" and "joan" are 1-typo away from "jofn". Similarly, in a prefix search, both "apple" and "apply" would match the prefix "app". The `default_sorting_field` tell Typesense which of the two equally-matched word that we should prioritize. That's why we insist of `default_sorting_field` -- by being a little opinionated.

I don't doubt that it's much more preferable and produces better results, it seems as though these are complex matters that you yourselves are handling. 

But I think this should be optional. As you said, the algorithm will use the insert order by default if all are constant. This should make it clear that the field `default_sorting_field` is already optional under the hood and not allowing the API to explicitly define it as optional is just adding an extra boundary to operate.

**Comment by kishorenc** [comment]
I agree with you and I've been intending to relook at this and make it an optional field. Thanks for the reminder.

**Comment by ashwinmram** [comment]
Hi Kishor, I agree with @stephanoparaskeva and hope you consider making this optional.  I've only included a default_sorting_field because it is mandatory and for no other reason.  The field I have selected will not help with ranking either because I used the tenant id which is common to all records for that particular tenant.  I do see the value of having a default sort, but it would be great if it was an optional configuration item.
